### PR TITLE
Create default solana acc when create\restore the wallet

### PR DESCRIPTION
**Pull request by cypt4**
Resolves https://github.com/brave/brave-browser/issues/25432

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
For desktop and android:
Enable BraveWalletSolana feature with create-default-solana-account = true
On android and desktop:
1) Create new wallet and check that Solana acc is created, default coin is Solana, and default network is Solana mainnet beta.
2) Restore wallet and check same conditions.
3) Check lock& restore case.
Enable BraveWalletSolana feature with create-default-solana-account = false
4) Check that there is no solana acc after wallet creation&restoration

### CODE REVIEW

**Comment by yrliou** [comment]
FYI, I saw there are test failures in solana_provider_browsertest.cc, you probably need to adjust the feature param there too as that file assumes no solana accounts were created.

File: components/brave_wallet/browser/brave_wallet_utils.h [hunk 1]
```diff
@@ -24,9 +24,12 @@ class GURL;
 
 namespace brave_wallet {
 
+extern const char kDefaultSolanaAccountName[];
```
Comment (darkdh) [inline components/brave_wallet/browser/brave_wallet_utils.h:27]: unused?

File: browser/brave_wallet/keyring_service_unittest.cc [hunk 2]
```diff
@@ -1565,6 +1584,38 @@ TEST_F(KeyringServiceUnitTest, CreateAndRestoreWallet) {
   EXPECT_FALSE(observer.IsKeyringRestored(mojom::kSolanaKeyringId));
 }
 
+TEST_F(KeyringServiceUnitTest, DefaultSolanaAccountCreated) {
+  KeyringService service(json_rpc_service(), GetPrefs());
```
Comment (darkdh) [inline browser/brave_wallet/keyring_service_unittest.cc:1588]: Why does this not need `create-default-solana-account` to be true?

File: browser/brave_wallet/brave_wallet_p3a_unittest.cc [hunk 3]
```diff
@@ -105,6 +107,16 @@ TEST_F(BraveWalletP3AUnitTest, DefaultSolanaWalletSetting) {
 }
 
 TEST_F(BraveWalletP3AUnitTest, KeyringCreated) {
+  base::test::ScopedFeatureList feature_list;
+  base::FieldTrialParams parameters;
+  parameters["create-default-solana-account"] = "false";
```
Comment (darkdh) [inline browser/brave_wallet/brave_wallet_p3a_unittest.cc:112]: can we use `brave_wallet::features::kCreateDefaultSolanaAccount.name`?

File: components/brave_wallet/common/features.cc [hunk 4]
```diff
@@ -23,6 +23,8 @@ const base::Feature kBraveWalletFilecoinFeature{
 
 const base::Feature kBraveWalletSolanaFeature{"BraveWalletSolana",
                                               base::FEATURE_ENABLED_BY_DEFAULT};
+const base::FeatureParam<bool> kCreateDefaultSolanaAccount{
+    &kBraveWalletSolanaFeature, "create-default-solana-account", true};
```
Comment (darkdh) [inline components/brave_wallet/common/features.cc:27]: What should the naming convention be here? `CreateDefaultSolanaAccount` or `create_default_solana_account` or `create-default-solana-account`?

File: browser/brave_wallet/keyring_service_unittest.cc (hunk 2 above)
Comment (cypt4) [reply browser/brave_wallet/keyring_service_unittest.cc:1588]: It is true by default

File: browser/brave_wallet/brave_wallet_p3a_unittest.cc (hunk 3 above)
Comment (cypt4) [reply browser/brave_wallet/brave_wallet_p3a_unittest.cc]: Fixed

File: components/brave_wallet/browser/brave_wallet_utils.h (hunk 1 above)
Comment (cypt4) [reply components/brave_wallet/browser/brave_wallet_utils.h]: Fixed

File: components/brave_wallet/common/features.cc (hunk 4 above)
Comment (cypt4) [reply components/brave_wallet/common/features.cc]: Duno, i've seen CamelCase and both "-" and "_" separated param names both in brave and chromium code, seems there is no naming conventions here. Changed to "_" cause it happens more often.

File: browser/brave_wallet/keyring_service_unittest.cc (hunk 2 above)
Comment (cypt4) [reply browser/brave_wallet/keyring_service_unittest.cc:1580]: Removed this

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 12 running 1.45.64 x64 build 
- Verified steps from PR
- Verified Solana network is set as active network for new wallet or restored wallet
- Verified first Solana account generated for the seed is set as the active account for new wallet or restored wallet
- Verified on upgrade from previous build, Solana isn't set as active network
- Verified on upgrade from previous build with Solana account created doesn't make it the active account unless it was set as active before upgrade
- Verified no crash when creating new wallet
- Verified Solana account is selected after account creation

https://user-images.githubusercontent.com/17010094/191344985-ad392ae6-a59f-436c-843a-5fb09d2888cd.mp4

**Pull request merged**